web/src/containers/activity/Goals.js

```diff
@@ -1,61 +1,133 @@
+import { Button, Review, TextField } from '@cmsgov/design-system-core';
+
 import PropTypes from 'prop-types';
-import React, { Component, Fragment } from 'react';
+import React, { Component, Fragment, useState } from 'react';
 import { connect } from 'react-redux';
 
 import {
   addActivityGoal as addActivityGoalAction,
   removeActivityGoal as removeActivityGoalAction,
   updateActivity as updateActivityAction
 } from '../../actions/activities';
-import Btn from '../../components/Btn';
-import CollapsibleList from '../../components/CollapsibleList';
-import { RichText } from '../../components/Inputs';
-import Instruction from '../../components/Instruction';
 import NoDataMsg from '../../components/NoDataMsg';
 import { Subsection } from '../../components/Section';
 import { t } from '../../i18n';
 
-const plaintext = txt => {
-  const e = document.createElement('div');
-  e.innerHTML = txt;
-  return e.innerText;
+const GoalReview = ({ goal, idx, edit, handleDelete }) => (
+  <Review
+    heading={
+      <h5 className="ds-h4">
+        {idx + 1}. {goal.description}
+      </h5>
+    }
+    editHref="#"
+    onEditClick={edit}
+    editContent={
+      <div className="nowrap">
+        <Button size="small" variation="transparent" onClick={edit}>
+          Edit
+        </Button>
+        {handleDelete && (
+          <Fragment>
+            |
+            <Button size="small" variation="transparent" onClick={handleDelete}>
+              Remove
+            </Button>
+          </Fragment>
+        )}
+      </div>
+    }
+  >
+    <p className="ds-u-margin-top--2">Benchmark: {goal.objective}</p>
+  </Review>
+);
+
+GoalReview.propTypes = {
+  edit: PropTypes.func.isRequired,
+  goal: PropTypes.object.isRequired,
+  handleDelete: PropTypes.func,
+  idx: PropTypes.number.isRequired
 };
 
-const GoalForm = ({ goal, idx, handleChange, handleDelete }) => (
-  <div className="mt2 mb3">
-    <Btn
-      kind="outline"
-      extraCss="right px-tiny py0 h5 xs-hide visibility--screen"
-      onClick={handleDelete}
-    >
-      ✗
-    </Btn>
-
-    <Instruction source="activities.goals.goal.instruction" />
-    <div className="mb3">
-      {t('activities.goals.goal.title', { number: idx + 1 })}
-      <RichText
-        content={goal.description}
-        onSync={handleChange(idx, 'description')}
-      />
-    </div>
-
-    <Instruction source="activities.goals.objective.instruction" />
-    <div className="mb3">
-      {t('activities.goals.objective.title')}
-      <RichText
-        content={goal.objective}
-        onSync={handleChange(idx, 'objective')}
-      />
-    </div>
+GoalReview.defaultProps = {
+  handleDelete: null
+};
+
+const GoalForm = ({ goal, idx, done, handleChange }) => (
+  <div className="ds-u-padding-bottom--2 ds-u-border-bottom--2">
+    <TextField
+      name="name"
+      className="data-entry-box"
+      label="Goal name"
+      hint={t('activities.goals.goal.instruction.detail', {
```

Comment:
Not wild about coopting the `instruction` content structure here. Should we also have an `input` structure?

```yaml
input:
  label: Goal name
  hint: Write your goal name here, obviously
```

I don't know.  Our YAML situation is kind of a mess, with some stuff being there and other stuff not...

Reply:
It's not especially helpful to just say 💯 here but 💯. My own take is:

* Content of any significance should be in YAML. It's pretty rough going back and forth.
* YAML things should have semantic meaning to the editors. `instruction` should be about telling people what to do and, if they are confused, how to do it. It should not govern interactive things. If we have inputs, we should have something else. `input` if that's it; something else if we want to govern, say, inputs and widgets and games of hangman. This is doubly true because the YAML doesn't correspond directly to the structure of the site.

Reply:
👌 I've changed the YAML.  I added a key for each field, and put an input key under those, like this:

```
objective:
  input:
    label: Benchmarks
    hint: >-
      Briefly describe how the state will measure progress toward the identified goal.
```

I also took the "Add a goal" button text out of the YAML, but happy to put it back if you think it belongs.  I struggle with what qualifies as having "significance."  I assume things like button text generally don't, and it's more about text that describes what something is or how to use it.

At one time, ***all*** of the user-facing text was in YAML, but that seemed like such overkill, and it was hard to keep track of too.  Finding the middleground is proving difficult (to me, at least), too.  😬